src/python/pants/build_graph/mutable_build_graph.py

```diff
@@ -74,14 +76,21 @@ def inject_address_closure(self, address):
 
       if not self.contains_address(target_address):
         target = self._target_addressable_to_target(target_address, target_addressable)
+        target.apply_injectables(self)
         self.inject_target(target, dependencies=dep_addresses)
       else:
         for dep_address in dep_addresses:
           if dep_address not in self.dependencies_of(target_address):
             self.inject_dependency(target_address, dep_address)
         target = self.get_target(target_address)
 
-      for traversable_spec in target.traversable_dependency_specs:
+      traversables = [target.compute_dependency_specs(payload=target.payload)]
+      # Only poke `traversable_dependency_specs` if a concrete implementation is defined
+      # in order to avoid spurious deprecation warnings.
```

Comment:
Hm... speaking of which... how will someone defining `traversable_dependency_specs` get that warning?

Should there be an explicit `deprecated_conditional` or similar here to confirm that it is _not_ defined?